File: gitlfs-server/src/main/java/ru/bozaro/gitlfs/server/ServerError.java
package ru.bozaro.gitlfs.server;

import jakarta.servlet.http.HttpServletResponse;

import javax.annotation.Nonnull;

/**
 * Server side error exception.
 *
 * @author Artem V. Navrotskiy
 */
public class ServerError extends Exception {
  private final int statusCode;

  public ServerError(int statusCode, String message) {
    super(message);
    this.statusCode = statusCode;
  }

  public ServerError(int statusCode, String message, Throwable cause) {
    super(message, cause);
    this.statusCode = statusCode;
  }

  public int getStatusCode() {
    return statusCode;
  }

  public void updateHeaders(@Nonnull HttpServletResponse response) {
  }
}

File: gitlfs-server/src/main/kotlin/ru/bozaro/gitlfs/server/ServerError.kt
package ru.bozaro.gitlfs.server

import jakarta.servlet.http.HttpServletResponse

/**
 * Server side error exception.
 *
 * @author Artem V. Navrotskiy
 */
open class ServerError : Exception {
    val statusCode: Int

    constructor(statusCode: Int, message: String?) : super(message) {
        this.statusCode = statusCode
    }

    constructor(statusCode: Int, message: String?, cause: Throwable?) : super(message, cause) {
        this.statusCode = statusCode
    }

    open fun updateHeaders(response: HttpServletResponse) {}
}
